### DISCUSSION TITLE
Auto create personal company

### DISCUSSION THREAD

**Discussion by loveshpatni** [andrewdwallo/filament-companies#114] [Q&A] [answered]
Is it possible to automatically create the personal company after the sign-up process?

**Comment by andrewdwallo** [comment, marked as answer]
Yes, you will need to override the [Register](https://github.com/andrewdwallo/filament-companies/blob/31bbf1f4384aea1c5f774c23ccbf7786d4dc7b4d/src/Pages/Auth/Register.php) page as well as the underlying flow of the default Filament registration page `Filament\Pages\Auth\Register`. Of course, you can create a custom registration page, but you will still need to override some things in the Filament registration page.

When registering through Socialite with this package, a personal company is automatically created as well. For guidance, you can refer to the registration process in the [OAuthController](https://github.com/andrewdwallo/filament-companies/blob/31bbf1f4384aea1c5f774c23ccbf7786d4dc7b4d/src/Http/Controllers/OAuthController.php#L260), which provides an example of what to implement on your registration page. The [CreateUserFromProvider](https://github.com/andrewdwallo/filament-companies/blob/31bbf1f4384aea1c5f774c23ccbf7786d4dc7b4d/stubs/app/Actions/FilamentCompanies/CreateUserFromProvider.php) action is used in this file, so you might also find the [CreateNewUser](https://github.com/andrewdwallo/filament-companies/blob/31bbf1f4384aea1c5f774c23ccbf7786d4dc7b4d/stubs/app/Actions/FilamentCompanies/CreateNewUser.php) action useful for your needs.

**Reply by loveshpatni** [reply]
Thanks, Andrew! I will follow these steps.

**Reply by andrewdwallo** [reply]
Your welcome!
